src/navigation/gate.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+from typing import ClassVar
+from unicodedata import normalize
+from navigation.context import Post
+
+import numpy as np
+
+from context import Context, Environment
+from state import BaseState
+from dataclasses import dataclass
+from drive import get_drive_command
+from util.np_utils import normalized, perpendicular_2d
+
+STOP_THRESH = 0.2
+DRIVE_FWD_THRESH = 0.95
+
+
+@dataclass
+class GateTrajectory:
```

Comment:
inheritance please

Reply:
ok I pushed a change that should address this